src/main/java/com/google/devtools/build/lib/packages/SkylarkNativeModule.java

```diff
@@ -118,7 +119,18 @@
       throw new EvalException(loc, "illegal argument in call to glob", e);
     }
 
-    return StarlarkList.copyOf(thread.mutability(), matches);
+    String[] result = new String[matches.size()];
```

Comment:
I'm a bit afraid of moving the sorting out of PackageFunction and into here, since it invites a future bug where `glob` evaluation goes through something other than SkylarkNativeModule. @c-parsons thoughts?

@meisterT are you sure c4f2d80270f1ce947fcf7fb0a4e5f0afb3a7062d needed to add support for the '@' character? In general, I wish Bazel supported less stuff...

Reply:
I have no personal opinion about whether `@` should be supported or not. (One might note that #374 has more thumbs up and advocacy that the average Bazel issue.) I expect at this point, we couldn't delete this behavior with a deprecation period.

I agree removing sorting from `PackageFunction` isn't unproblematic. But as far as fixing the linked issue is concerned, to not sort here, we'd have to think of some other place shared by legacy and skyframe globbing to put the '@'-prefixing logic. I also don't conceptually think this prefixing behavior belongs to globbing. Globbing should return filesystem paths not targets.

Reply:
> I expect at this point, we couldn't delete this behavior with a deprecation period.

Yes, ofc.

> I also don't conceptually think this prefixing behavior belongs to globbing. Globbing should return filesystem paths not targets.

Completely agree.

I was curious about the '@' support b/c if it hypothetically didn't exist then we wouldn't be talking about how the inappropriate location of the prefixing code.

Reply:
`@` is quite common in npm package names, e.g. [`@bazel/typescript`](https://www.npmjs.com/package/@bazel/typescript), and consequently shows up in filenames (particularly as a directory name...)

Reply:
Yes, npm was the driving factor behind my change.

Reply:
Okay. How about the following:
- Rename Globber#fetch to Globber#fetchUnsorted (and update javadoc accordingly). Your changes in this CL are making the javadoc for Globber#fetch not match either implementation, and I also don't want future code readers to assume the returned List<String> has any meaningful order.
- Keep your colon-prefixing and sorting here in SkylarkNativeModule.java. I really like your reasoning that colon-prefixing doesn't belong in the core globbing code.